cabal-install/src/Distribution/Client/Init/Types.hs

```diff
@@ -455,7 +455,13 @@ newtype BreakException = BreakException String deriving (Eq, Show)
 instance Exception BreakException
 
 -- | Used to inform the intent of prompted messages.
-data Severity = Log | Info | Warning | Error deriving (Eq, Show)
+data Severity = Info | Warning | Error deriving (Eq)
+
+displaySeverity :: Severity -> String
+displaySeverity severity
+  | severity == Info = "Info"
+  | severity == Warning = "Warn"
+  | severity == Error = "Err"
```

Comment:
this is a prime example of when to use a [Pattern Match](https://en.wikibooks.org/wiki/Haskell/Pattern_matching)! I would suggest the following: 

 ```suggestion
displaySeverity severity = case severity of 
   | Info -> "Info"
   | Warning -> "Warn"
   | Error -> "Err"
```

This is equivalent, but with more idiomatic code. It also doesn't involve `Eq`. Equivalently, you could also write: 

```haskell
displaySeverity :: Severity -> String
displaySeverity Info = "Info"
displaySeverity Warning = "Warn" 
displaySeverity Error = "Err"
```

Your choice.

Reply:
I chose case expression because I often navigate using search. Repeating words make this way less convenient.

Reply:
One more step I'd suggest is to consider using the `LambdaCase` GHC extension, which allows:
```
displaySeverity = \case 
  | Info -> "Info"
  | Warning -> "Warn"
  | Error -> "Err"
```

Reply:
That's neat, I'll add the suggested change.

Thanks.

Reply:
Oh, my bad, sorry. I'm happy with either version tbh.